### PR TITLE
zli BastionZero cli 4.9.0

### PR DESCRIPTION
**Pull request by Nabeelperson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/zli.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class Zli < Formula
+  desc "BastionZero cli"
+  homepage "https://www.bastionzero.com"
+  url "https://github.com/bastionzero/zli/releases/download/4.9.0/zli-4.9.0.tar.gz"
+  sha256 "553b7a82a1b4b5015acc9bbe645afcca1fb351e92dbfe6514d7355e8df145136"
+  license "Apache-2.0"
+  head "https://github.com/bastionzero/zli.git"
+
+  bottle :unneeded
```
Comment (carlocab) [inline Formula/zli.rb:11]: Why is this `bottle :unneeded`?

File: Formula/zli.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  depends_on "node@12"
+
+  def install
+    system "npm", "install", *Language::Node.local_npm_install_args
+    system "npm", "run", "release"
+    rm "./bin/zli-win.exe"
+    rm "./bin/zli-linux"
+    mv "./bin/zli-macos", "./bin/zli"
+    bin.install "./bin/zli"
```
Comment (carlocab) [inline Formula/zli.rb:20]:
```suggestion
    bin.install "bin/zli-macos" => "zli"
```

File: Formula/zli.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  test do
+    system "zli", "help"
```
Comment (carlocab) [inline Formula/zli.rb:24]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @Nabeelperson.

File: Formula/zli.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class Zli < Formula
+  desc "BastionZero cli"
+  homepage "https://www.bastionzero.com"
+  url "https://github.com/bastionzero/zli/releases/download/4.9.0/zli-4.9.0.tar.gz"
+  sha256 "553b7a82a1b4b5015acc9bbe645afcca1fb351e92dbfe6514d7355e8df145136"
+  license "Apache-2.0"
+  head "https://github.com/bastionzero/zli.git"
+
+  bottle :unneeded
+  depends_on "node@12"
```
Comment (carlocab) [inline Formula/zli.rb:12]: Can this depend on `node` instead of `node@12`?

File: Formula/zli.rb (hunk 1 above)
Comment (Nabeelperson) [reply Formula/zli.rb:11]: Not sure, I think this is an artifact of me learning how to write a brew formula. Will remove.

**Comment by carlocab** [comment]
This looks like a nice piece of software, but it currently does not meet the requirements for inclusion in homebrew/core. The CI log reads
```
zli:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 1 problem in 1 formula detected
```
You may choose to submit this again when this requirement is met.

If you wish, you can host this formula in your own tap in the meantime. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Comment by Nabeelperson** [comment]
@carlocab Thank you for your prompt and thorough review, my team and I really appreciate it. We will move forward with hosting the formula in our own tap.

**Pull request closed without merging**